lib/index.ts

```diff
@@ -48,10 +52,26 @@ class TornAPI {
         }
     }
 
+    private comment = '';
+    public setComment(comment: string): void {
+        // only reset if the comment has changed
+        if (this.comment !== comment) {
+            this.comment = comment;
+
+            // when the comment changes, reset all the internals
+            this._torn = null;
+            this._user = null;
+            this._faction = null;
+            this._property = null;
+            this._itemmarket = null;
+            this._company = null;
```

Comment:
Can we extract this from here and `setKey` into its own routine?

Reply:
Just for more info, `setKey` and `setComment` functions are almost identical. I would like to remove the following from each and put it into its own routine that is called from both the setters.

 ```
            // when the comment changes, reset all the internals
            this._torn = null;
            this._user = null;
            this._faction = null;
            this._property = null;
            this._itemmarket = null;
            this._company = null;
```

So maybe ...

```
private resetInternals(): void {
  this._torn = null;
  this._user = null;
  this._faction = null;
  this._property = null;
  this._itemmarket = null;
  this._company = null;
}
```